### ISSUE TITLE
Changing Pod name through metadata can cause 2 pods to be created

### ISSUE THREAD

**Issue by jthomperoo** [jthomperoo/custom-pod-autoscaler-operator#95]
**Describe the bug**
Change introduced in #93 means that you can modify the pod name independently of the CPA name, this causes issues because you can change the name of the pod, so the old pod won't get any reconcile event, meaning it wont be deleted and you will end up with two pods.

**To Reproduce**
1. Deploy a CPA with some metadata for the Pod name:

```yaml
spec:
  template:
    metadata:
      name: test-pod
...
```

2. Wait and see the Pod is created.
3. Update the metadata to change the Pod name:

```yaml
spec:
  template:
    metadata:
      name: test-pod-2
...
```

4. Wait and see that there are now 2 pods.

**Expected behavior**
The old pod should be deleted as the new one is created.

**Kubernetes Details (`kubectl version`):**

```
v1.22.7
```

**Custom Pod Autoscaler Operator Details:**
Unreleased CPA based off `master` (commit [`1efee1a5076b20aaccb467b39314fcc2fae68ec4`](https://github.com/jthomperoo/custom-pod-autoscaler-operator/commit/1efee1a5076b20aaccb467b39314fcc2fae68ec4))

**Additional context**
N/A

**Comment by StefanCenusa** [comment]
You could create a deployment to manage your pod (this might also help in situations where cluster autoscaler is blocked by pods without controllers - although there is an annotation that can be added to the pod to be evicted) with Recreate update policy, thus ensuring that there will be no instances running in parallel.
Or, the alternative, CPA could be the pod's controller, manage it through label selectors (as StatefulSet controller does to its pods)

**Comment by jthomperoo** [comment]
That's a really good idea, using a Deployment might be a much better approach to this; I'll try that out this weekend and see if that works, think that could solve this for me! Thanks!

**Comment by StefanCenusa** [comment]
But then the name of the deployment could also lead to the same problem (having a dangling old deployment) If that would be configurable through the CR.

**Comment by StefanCenusa** [comment]
It depends on how the CRD will look. As long as I can pass some metadata / spec to the pod, I can work with it. 

I don't think you can configure Pod's names through deployment spec (they are build from deployment name + replica set id + random id).

**Comment by jthomperoo** [comment]
Your suggestion of adding a label to link the Pod back to the CPA was a good idea. I've set up a pull request (#96) to introduce this change, with every resource now tracked by the `v1.custompodautoscaler.com/owned-by` label with its value set to the CPA name. 

This allows for each reconcile loop the operator can get all Pods with that label, and if the name doesn't match the way it's configured in the `PodTemplateSpec` in the CPA definition it will delete it.

I'll finish up testing on this and get a new version cut today :)

Thanks for the feedback + helping out with designing this!

**Comment by jthomperoo** [comment]
Now available in [`1.3.0`](https://github.com/jthomperoo/custom-pod-autoscaler-operator/releases/tag/v1.3.0) 🎉
